pyedr/pyedr/pyedr.py

```diff
@@ -276,14 +299,74 @@ def do_enx(self):
                 try:
                     sub.val = ndo_readers[sub.type](data, sub.nr)
                 except IndexError:
-                    raise ValueError("Reading unknown block data type: this file is corrupted or from the future")
+                    raise ValueError("Reading unknown block data type: "
+                                     "this file is corrupted "
+                                     "or from the future")
+
+    def convert_full_sums(self):
+        """Convert old energy sums
+
+        Old energy files seem to store the sums
+        over all preceding energy frames.
+        Therefore one has to calculate the difference between frames.
+        See convert_full_sums in enxio.cpp
+        """
+        fr = self.frame
+
+        first_step = self.first_step
+        nsum_prev = self.nsum_prev
+        step_prev = self.step_prev
+        ener_prev = self.ener_prev
+
+        if fr.nsum > 0:
+            ne = ns = 0
+            for ener in fr.ener:
+                if ener.e != 0:
+                    ne += 1
+                if ener.esum != 0:
+                    ns += 1
+            if ne > 0 and ns == 0:
+                # We do not have all energy sums
+                fr.nsum = 0
```

Comment:
I'm ignoring a lot of the other uncovered pieces of code since it's historical poor test coverage, but here and line 351 seems to be new code. Is there a way to test these branches? Possibly using mock / monkeypatch?

Reply:
I've added some mock files which add ~5 KB of new test data (+ 5 KB of mock-generation script). Most of the new files are malformed (invalid magic etc) and should trigger most warnings and errors. Also, all of the legacy code paths are covered (and it seems nobody has ever tried to read char or int64 sub blocks).
Some errors/warnings are logically not reachable and were replaced by `assert`s.